### PR TITLE
Introduce unit tests and refactor web dart:ui into "package"

### PR DESCRIPTION
**Pull request by jonahwilliams**
The package will be transformed via some trivial rules into a dart:ui implementation. This allows the web engine to have unit tests provided by build_runner's `pub run build_runner test -- -p chrome`

Waiting on other PR and pending work to transform back.

cc @yjbanov

### CODE REVIEW

**Comment by yjbanov** [comment]
Interesting. So you don't need any conversion between the package structure and `dart:*` structure, you just use a least common denominator layout?

**Comment by jonahwilliams** [comment]
Not quite.. I need to regex replace ui with dart.ui, which I think

**Comment by jonahwilliams** [comment]
With this change I rewrite the `ui` declaration into a `dart.ui` declaration, fairly simple for now. I can also add logic to hide specific classes.

File: web_sdk/sdk_rewriter.dart [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (yjbanov) [inline web_sdk/sdk_rewriter.dart]: 2019?

**Pull request merged**